src/signature_mahalanobis_knn/sig_mahal_knn.py

```diff
@@ -5,117 +5,179 @@
 from joblib import Parallel, delayed
 from sklearn.metrics import roc_auc_score
 from sklearn.neighbors import NearestNeighbors
-from sktime.transformations.panel.signature_based import SignatureTransformer
 
 from signature_mahalanobis_knn.mahal_distance import Mahalanobis
+from signature_mahalanobis_knn.utils import plot_roc_curve
 
 
 class SignatureMahalanobisKNN:
     def __init__(
         self,
-        n_jobs=-2,
-        augmentation_list=("addtime",),
-        window_name="global",
-        window_depth=None,
-        window_length=None,
-        window_step=None,
-        rescaling=None,
-        sig_tfm="signature",
-        depth=2,
+        n_jobs: int = -2,
     ):
         """
-        __description__
-
-        :param n_jobs: parameter for joblib, number of parallel processors to use.
-        :param augmentation_list: Possible augmentation strings are
-        ['leadlag', 'ir', 'addtime', 'cumsum', 'basepoint']
-        :param window_name: str, String from ['global', 'sliding', 'expanding', 'dyadic']
-        :param window_depth: int, The depth of the dyadic window. (Active only if
-        `window_name == 'dyadic']`.
-        :param window_length: int, The length of the sliding/expanding window. (Active
-        only if `window_name in ['sliding, 'expanding'].
-        :param window_step: int, The step of the sliding/expanding window. (Active
-        only if `window_name in ['sliding, 'expanding'].
-        :param rescaling: "pre" or "post",
-                    "pre": rescale the path last signature term should be roughly O(1)
-                    "post": Rescals the output signature by multiplying the depth-d term by d!.
-                            Aim is that every term become ~O(1).
-        :param sig_tfm: One of: ['signature', 'logsignature']).
-        :param depth: int, Signature truncation depth.
```

Comment:
I would prefer to keep these descriptions somewhere, maybe in the documentation, as they are not obvious (even for me.... not to mention the user).

Reply:
it's a bit non-standard to repeat another codebase's documentation, but happy to add them back. will do this tomorrow

Reply:
added them back in latest commit